src/app/components/task-details/task-details.component.html

```diff
@@ -1,9 +1,8 @@
-@if (this.task) {
-    <div class="container">
-        <app-task [task]="this.task"></app-task>
-    </div>
-} 
-@else {
-        <h1>A task with the id: "{{this.id}}" doesn't exist!</h1>
-}
-
+<div class="card">
+    <a class="close-form" routerLink="/">x</a>
+    <h1 class="dialog-title">Edit task</h1>
+    <!-- Only creating the form when we have task, otherwise validator dones't get task title -->
+    @if(this.task){
```

Comment:
you shouldn't have this check, if the task doesn't exist then it should not be possible to go to the details page